server/routes/api/messages.js

```diff
@@ -9,28 +9,30 @@ router.post("/", async (req, res, next) => {
       return res.sendStatus(401);
     }
     const senderId = req.user.id;
-    const { recipientId, text, conversationId, sender } = req.body;
+    const sender = req.user;
+    const { recipientId, text } = req.body;
 
-    // if we already know conversation id, we can save time and just add it to message and return
-    if (conversationId) {
-      const message = await Message.create({ senderId, text, conversationId });
-      return res.json({ message, sender });
-    }
     // if we don't have conversation id, find a conversation to make sure it doesn't already exist
     let conversation = await Conversation.findConversation(
       senderId,
       recipientId
     );
 
-    if (!conversation) {
-      // create conversation
-      conversation = await Conversation.create({
-        user1Id: senderId,
-        user2Id: recipientId,
-      });
-      if (onlineUsers.includes(sender.id)) {
-        sender.online = true;
-      }
+    // if we already know conversation id, we can save time and just add it to message and return
+    if (conversation) {
+      const conversationId = conversation.id;
+      let message = await Message.create({ senderId, text, conversationId });
+      //return the sender
+      return res.json({ message, sender:null });
+    }
+
+    // create conversation
+    conversation = await Conversation.create({
+      user1Id: senderId,
+      user2Id: recipientId,
+    });
+    if (onlineUsers.includes(sender.id)) {
+      sender.online = true;
```

Comment:
Same as above :slightly_smiling_face: (changes unrelated to the issue being addressed)